extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java

```diff
@@ -1132,33 +1223,39 @@ private JsonObject getVideoSecondaryInfoRenderer() throws ParsingException {
             throw new ParsingException("Could not find videoSecondaryInfoRenderer");
         }
 
-        this.videoSecondaryInfoRenderer = theVideoSecondaryInfoRenderer;
+        videoSecondaryInfoRenderer = theVideoSecondaryInfoRenderer;
         return theVideoSecondaryInfoRenderer;
     }
 
     @Nonnull
-    private Map<String, ItagItem> getItags(final String streamingDataKey,
-                                           final ItagItem.ItagType itagTypeWanted) {
+    private Map<String, ItagItem> getItags(@Nonnull final String streamingDataKey,
+                                           @Nonnull final ItagItem.ItagType itagTypeWanted) {
         final Map<String, ItagItem> urlAndItags = new LinkedHashMap<>();
-        if (desktopStreamingData == null && mobileStreamingData == null) {
+        if (desktopStreamingData == null && androidStreamingData == null) {
             return urlAndItags;
         }
 
-        // Use the mobileStreamingData object first because there is no n param and no
+        // Use the androidStreamingData object first because there is no n param and no
         // signatureCiphers in streaming URLs of the Android client
         urlAndItags.putAll(getStreamsFromStreamingDataKey(
-                mobileStreamingData, streamingDataKey, itagTypeWanted));
+                androidStreamingData, streamingDataKey, itagTypeWanted, androidCpn));
+        urlAndItags.putAll(getStreamsFromStreamingDataKey(
+                desktopStreamingData, streamingDataKey, itagTypeWanted, desktopCpn));
+        // Use the iosStreamingData object in the last position because most of the available
+        // streams can be extracted with the Android and web clients and also because the iOS
+        // client is only enabled by default on livestreams
         urlAndItags.putAll(getStreamsFromStreamingDataKey(
-                desktopStreamingData, streamingDataKey, itagTypeWanted));
+                iosStreamingData, streamingDataKey, itagTypeWanted, androidCpn));
```

Comment:
Also 3 times the same code here...
Maybe use a map and a loop or the java streams api?

Reply:
> `iosStreamingData, streamingDataKey, itagTypeWanted, androidCpn`

Thank you for selecting this line in your review, you let me found a bug in it!

-> `androidCpn` should be `iosCpn`.

Reply:
Done, I used a HashMap.